limiter.go

```diff
@@ -71,6 +71,7 @@ func (dl *dialLimiter) finishedDial(dj *dialJob) {
 
 		if len(dl.waitingOnFd) > 0 {
 			next := dl.waitingOnFd[0]
+			dl.waitingOnFd[0] = nil // clear out memory
```

Comment:
This is really the main issue. If we build a massive backlog and then stop dialing, we won't free *anything* till we're done clearing it unless we do this.

Reply:
Is it because inaccessible objects from re-sliced slice are sill alive according to GC?

I didn't think this would be a case.